execute.py

```diff
@@ -0,0 +1,767 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Annealed Importance Sampling to exchange intramolecular talks between torchANI and openmm.
+
+import mdtraj as md
+import torch
+import torchani
+from openmmtools.constants import kB
+from simtk import unit
+
+atomic_num_to_symbol_dict = {1: 'H', 6: 'C', 7: 'N', 8: 'O'}
+mass_dict_in_daltons = {'H': 1.0, 'C': 12.0, 'N': 14.0, 'O': 16.0}
+import mdtraj.utils as mdtrajutils
+from openmmtools.states import ThermodynamicState, SamplerState
+
+from perses.dispersed.utils import configure_platform
+from openmmtools import cache, utils
+
+from perses.utils.openeye import smiles_to_oemol
+from openmoltools.forcefield_generators import generateTopologyFromOEMol
+from openmmforcefields.generators import SystemGenerator
+import simtk.openmm.app as app
+from openforcefield.topology import Molecule
+
+import numpy as np
+import os
+import simtk.openmm as openmm
+
+from scipy import optimize
+
+from perses.dispersed.utils import minimize
+
+cache.global_context_cache.platform = configure_platform(utils.get_fastest_platform().getName())
+
+import logging
+
+logging.basicConfig(level=logging.NOTSET)
+_logger = logging.getLogger("interpolANI")
+_logger.setLevel(logging.DEBUG)
+
+
+class ANI1_force_and_energy(object):
+    # some class attributes
+    mass_unit = unit.dalton
+    distance_unit = unit.nanometers
+    time_unit = unit.femtoseconds
+    energy_unit = unit.kilojoules_per_mole
+    ani_distance_unit = unit.angstroms
+    hartree_to_kJ_per_mole = 2625.499638
+    ani_energy_unit = hartree_to_kJ_per_mole * energy_unit
+    nm_to_angstroms = 10.
+    angstroms_to_nm = 1e-1
+
+    def __init__(self,
+                 model,
+                 atoms,
+                 platform='cpu',
+                 temperature=300 * unit.kelvin
+                 ):
+
+        """
+        Performs energy and force calculations.
+        Slightly modified code from:
+            https://gist.github.com/wiederm/7ac5c29e5a0dea9d17ef16dda93fe02d#file-reweighting-py-L42; thanks, Marcus
+
+        Parameters
+        ----------
+        model: torchani.models object
+            model from which to compute energies and forces
+        atoms: str
+            a string of atoms in the indexed order
+        platform : str, default 'cpu',
+            platform on which to initialize the model device
+        temperature : float * unit.kelvin, default 300 * unit.kelvin
+            temperature
+        """
+        self.model = model
+        self.atoms = atoms
+
+        self.platform = platform
+        self.device = torch.device(self.platform)
+        if self.platform == 'cpu':
+            torch.set_num_threads(2)
+        else:
+            raise Exception(f"we don't support gpu just yet")
+
+        self.species = self.model.species_to_tensor(atoms).to(self.device).unsqueeze(0)
+        self.temperature = temperature
+        self.beta = 1.0 / (kB * temperature)
+
+        self.W_shads = []
```

Comment:
Are these works and shadow-works?
would you mind documenting them pls

Reply:
let me delete this file; it's `complete.py` that is the one i use; im sorry